src/functions/ducklake_merge_adjacent_files.cpp

```diff
@@ -322,11 +322,31 @@ DuckLakeCompactor::GenerateCompactionCommand(vector<DuckLakeCompactionFileEntry>
 
 	// generate the LogicalGet
 	auto &columns = table.GetColumns();
+	auto table_path = table.DataPath();
+	string data_path;
+	if (!source_files.empty()) {
```

Comment:
Instead of looking at the data path directly, can we look at the `partition_id` and `partition_values` of the files